unstructured/file_utils/filetype.py

```diff
@@ -229,7 +232,11 @@ def detect_filetype(
             return EXT_TO_FILETYPE.get(extension, FileType.UNK)
 
     elif file is not None:
-        extension = None
+        try:
+            _, extension = os.path.splitext(file.name)
+        except AttributeError:
```

Comment:
Do you know what condition causes this exception to raise?

Reply:
It was one of the tests that passed in a _io.BytesIO object. I think it's `test_exploration.py` that initiates the test.

Reply:
Gotcha, that makes sense since this is in the `file` block. Is it the `name` attribute on file that triggers the exception? Would rather do

```python
if hasattr(file, "name'):
    _, extension = os.path.splitext(file.name)
else:
    extension = ""
```

in that case.

Reply:
Mainly to avoid masking a problem if there's another unrelated `AttributeError` that pops up down the line.

Reply:
It is. I'll make that change then